src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenLinkApplication.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.hmcts.divorce.citizen.event;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.divorce.common.model.CaseData;
+import uk.gov.hmcts.divorce.common.model.State;
+import uk.gov.hmcts.divorce.common.model.UserRole;
+import uk.gov.hmcts.divorce.solicitor.service.CcdAccessService;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+import static uk.gov.hmcts.divorce.common.model.State.AwaitingApplicant2Response;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_DIVORCE_SYSTEMUPDATE;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CITIZEN;
+import static uk.gov.hmcts.divorce.common.model.access.Permissions.CREATE_READ_UPDATE;
+
+@Slf4j
+@Component
+public class CitizenLinkApplication implements CCDConfig<CaseData, State, UserRole> {
+
+    @Autowired
+    private CcdAccessService ccdAccessService;
+
+    @Autowired
+    private HttpServletRequest httpServletRequest;
+
+    public static final String CITIZEN_LINK_APPLICANT_2 = "citizen-link-applicant2";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        configBuilder
+            .event(CITIZEN_LINK_APPLICANT_2)
+            .forState(AwaitingApplicant2Response)
+            .name("Link Applicant 2 to case")
+            .description("Link Applicant 2 to case to enable completion of joint application")
+            .grant(CREATE_READ_UPDATE, CITIZEN, CASEWORKER_DIVORCE_SYSTEMUPDATE)
```

Comment:
A citizen shouldn't be able to trigger this - it has to be the system.

Reply:
Will update this